lib/fourOhFour.js

```diff
@@ -0,0 +1,177 @@
+'use strict'
+
+const FindMyWay = require('find-my-way')
+
+const Reply = require('./reply')
+const Request = require('./request')
+const Context = require('./context')
+const {
+  kRoutePrefix,
+  kCanSetNotFoundHandler,
+  kFourOhFourLevelInstance,
+  kReply,
+  kRequest,
+  kContentTypeParser,
+  kBodyLimit,
+  kLogLevel,
+  kFourOhFourContext,
+  kFourOhFour,
+  kMiddlewares,
+  kHooks
+} = require('./symbols.js')
+
+function fourOhFour (logger, modifyCoreObjects, genReqId) {
+  // 404 router, used for handling encapsulated 404 handlers
+  const router = FindMyWay({ defaultRoute: fourOhFourFallBack })
+
+  const fof = {
+    router,
+    setNotFoundHandler: setNotFoundHandler,
+    setContext: setContext,
+    updateInstance: updateInstance
+  }
+
+  return fof
+
+  function updateInstance (instance) {
+    this[kFourOhFourLevelInstance] = instance
+  }
```

Comment:
Can you add some comments describing why this is needed?

Reply:
Sure, I will try to figure out better all the pieces in play